blockio-api/src/System/FS/BlockIO/API.hs

```diff
@@ -18,13 +18,22 @@ module System.FS.BlockIO.API (
   , ioopBufferOffset
   , ioopByteCount
   , IOResult (..)
+    -- * Advice
   , Advice (..)
+  , hAdviseAll
+  , hAdviseAll'
+  , hAdvisesAll
+  , hAdvisesAll'
+  , hDropCacheAll
+  , hDropCacheAll'
```

Comment:
As far as I can see, the only one of the `'` primed versions here that is used is `hDropCacheAll'`, and arguably that one should not be used either (replacing it with a use of `hDropCacheAll` just before closing the file).

So we could remove the variants that open the file just to apply advice (which itself only makes sense for the cache dropping advice, the other advice is local to the open file handle).

Also, `hAdvisesAll` is not used anymore and can be dropped.